syntax = "proto3";

package semio.server.v1;

import "model/v1/model.proto";

option go_package = "github.com/usalu/semio/src/schema/semio/translator/v1";

// The server service is the gateway for all other apis of semio.
service TranslatorService {
  //   option (google.api.default_host) = "localhost:50000";

  // Lay out a design from a layout and return a design.
  rpc TranslateRepresentation(semio.model.v1.Layout) returns (semio.model.v1.Representation) {
    // option (google.api.http) = {
    //   post: "v1/layout-design"
    //   body: "*"
    // };
  }
}

// Translate a representation from one pose (coordinate system) into another one.
message TranslateRepresentationRequest {
    semio.model.v1.Representation representation = 1;
    semio.model.v1.Pose target_pose = 2;
    semio.model.v1.Pose source_pose = 3;
}